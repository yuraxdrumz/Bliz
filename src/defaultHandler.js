let defaultHandler = function(req ,res, err){
  if(err){
    res.statusCode = err.status || 500
    res.json({error:err.toString()})

  }else{
    res.statusCode = 404
    res.json({error:`${req.method.toUpperCase()} - ${req.url} not found...`})
  }
}
export default defaultHandler