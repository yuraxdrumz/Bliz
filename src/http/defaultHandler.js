// default error handler handler
import { StructError } from '../main'

// default handler for http
// if error passed from handler fire it
// else if got here -> 404
const defaultErrorHandler = (req, res, err) => {
  if (err) {
    res.statusCode = err.status || 500
    if (err instanceof StructError) {
      res.json({
        error: err.message,
        path: err.path,
        dataPassed: err.data,
        valueReceived: err.value,
        typeExpected: err.type
      })
    } else {
      console.error(err)
      res.json({ error: err.message })
    }
  } else {
    res.statusCode = 404
    res.json({ error: `${req.method.toUpperCase()} - ${req.url} not found...` })
  }
}
export default defaultErrorHandler
