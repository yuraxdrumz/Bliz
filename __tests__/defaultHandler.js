import defaultHandler from '../src/defaultHandler'

const req = {
  method:'GET',
  url:'url/data/never/mind'
}

const res = {
  statusCode:null,
  json:jest.fn(data=>data)
}

describe('Default Handler suite', ()=>{
  test('should throw err if request and response not passed', ()=>{
    expect(()=>defaultHandler()).toThrowError()
  })
  test('should put statusCode to 404 if 3rd param error not passed', ()=>{
    expect(defaultHandler(req, res)).not.toBeDefined()
    expect(res.statusCode).toEqual(404)
    expect(res.json.mock.calls[0][0]).toMatchSnapshot()
  })
  test('should put statusCode to err status if 3rd param error was passed', ()=>{
    res.json.mockClear()
    const error = {
      status:700,
      toString:jest.fn(()=>'error to string')
    }
    expect(defaultHandler(req, res, error)).not.toBeDefined()
    expect(res.statusCode).toEqual(700)
    expect(res.json.mock.calls[0][0]).toMatchSnapshot()
  })
  test('should put statusCode to 500 if error status is defined and 3rd param error was passed', ()=>{
    res.json.mockClear()
    const error = {
      toString:jest.fn(()=>'error to string')
    }
    expect(defaultHandler(req, res, error)).not.toBeDefined()
    expect(res.statusCode).toEqual(500)
    expect(res.json.mock.calls[0][0]).toMatchSnapshot()
  })
})