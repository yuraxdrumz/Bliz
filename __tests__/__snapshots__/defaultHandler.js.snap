// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Handler suite should put statusCode to 404 if 3rd param error not passed 1`] = `
Object {
  "error": "GET - url/data/never/mind not found...",
}
`;

exports[`Default Handler suite should put statusCode to 500 if error status is defined and 3rd param error was passed 1`] = `
Object {
  "error": "error to string",
}
`;

exports[`Default Handler suite should put statusCode to err status if 3rd param error was passed 1`] = `
Object {
  "error": "error to string",
}
`;
